### PR TITLE
Build panic-semihosting for embedded target. Prepares v0.5.5. Closes #287.

### LINKED ISSUE #287: panic-semihosting docs are empty
https://docs.rs/panic-semihosting/0.5.4/panic_semihosting/ is empty, despite there being documentation on the crate.

Caused by `#![cfg(all(target_arch = "arm", target_os = "none"))]`.

The fix is to tell docs.rs to use an embedded ARM target (see https://docs.rs/about/metadata).

### PR DESCRIPTION
**Pull request by adamgreig**

### CODE REVIEW

**Review by jonas-schievink — APPROVED** [review]
bors r+

**Comment by adamgreig** [comment]
Thanks, published and tagged.

File: panic-semihosting/Cargo.toml [hunk 1]
```diff
@@ -20,3 +20,6 @@ cortex-m-semihosting = { path = "../cortex-m-semihosting", version = ">= 0.3, <
 exit = []
 inline-asm = ["cortex-m-semihosting/inline-asm", "cortex-m/inline-asm"]
 jlink-quirks = ["cortex-m-semihosting/jlink-quirks"]
+
+[package.emtadata.docs.rs]
```
Comment (adamgreig) [inline panic-semihosting/Cargo.toml:24]: Aw, nuts. https://docs.rs/panic-semihosting/0.5.5/panic_semihosting/

Comment (jonas-schievink) [reply panic-semihosting/Cargo.toml:24]: Aw, sorry for missing that

**Pull request merged**